pkg/platform/api/platform.go

```diff
@@ -0,0 +1,142 @@
+package api
+
+import (
+	"github.com/Masterminds/semver"
+	"github.com/bottlerocket-os/bottlerocket-update-operator/pkg/logging"
+	"github.com/bottlerocket-os/bottlerocket-update-operator/pkg/platform"
+
+	"github.com/pkg/errors"
+)
+
+// Assert Update-API as a platform implementor.
+var _ platform.Platform = (*Platform)(nil)
+
+type Platform struct {
+	log       logging.Logger
+	apiClient *apiClient
+}
+
+func New() (*Platform, error) {
+	return &Platform{log: logging.New("platform"), apiClient: newAPIClient()}, nil
+}
+
+type statusResponse struct {
+	osVersion *semver.Version
+}
+
+func (sr *statusResponse) OK() bool {
+	// Bottlerocket OS version needs to be at least a certain version to support the Update API
+	constraint, _ := semver.NewConstraint(">= " + minimumRequiredOSVer)
```

Comment:
nit: the error should be checked and the object reused to keep from parsing the constraint's expression each time.